### PR TITLE
Repositories as traits

### PR DESCRIPTION
**Pull request by LucasHazardous**
All tests pass except for "composition_fetching_by_page" but it didn't pass before my changes anyway.

### CODE REVIEW

**Review by Dudu7352 — APPROVED** [review]

**Comment by lechos22** [comment]
Didn't notice that before, but it seems to be my fault, so I'll fix it now

File: src/database/data_access/composition.rs [hunk 1]
```diff
@@ -22,33 +22,47 @@ use crate::database::{
...
-    let page: Option<Composition> = connection.select(&page.composition).await?;
-    match page {
-        Some(page) => Ok(page),
-        None => Err(DatabaseError::RecordNotFound),
-    }
+pub trait CompositionRepository {
+    fn get_composition_from_page(
+        &self,
+        page: &Page,
+    ) -> impl std::future::Future<Output = Result<Composition, DatabaseError>> + Send;
```
Comment (lechos22) [inline src/database/data_access/composition.rs:29]: Why not add Future to imports to increase readability?

**Review by lechos22 — CHANGES_REQUESTED** [review]
Why not add Future to imports to increase readability?

**Comment by lechos22** [comment]
Other than that, everything looks good

**Comment by LucasHazardous** [comment]
Changes applied. Thanks for fixing tests.

**Pull request merged**